app/controllers/api/hosts_controller.rb

```diff
@@ -15,8 +15,9 @@ def edit_resource(type, id, data = {})
       resource_search(id, type, collection_class(:hosts)).tap do |host|
         all_credentials = Array.wrap(credentials).each_with_object({}) do |creds, hash|
           auth_type = creds.delete(AUTH_TYPE_ATTR) || DEFAULT_AUTH_TYPE
+          creds.symbolize_keys!
           creds.reverse_merge!(:userid => host.authentication_userid(auth_type))
-          hash[auth_type.to_sym] = creds.symbolize_keys!
+          hash[auth_type.to_sym] = creds
```

Comment:
I'd rather prefer not using mutability here and drop the exclamation marks:
```suggestion
          hash[auth_type.to_sym] = creds.symbolize_keys.reverse_merge(:userid => host.authentication_userid(auth_type))
```

Reply:
I just following existing pattern, trying to KISS